src/NServiceBus.Core/Performance/Metrics/MessagingMetricsFeature.cs

```diff
@@ -0,0 +1,39 @@
+namespace NServiceBus
+{
+    using System;
+    using Features;
+
+    /// <summary>
+    /// MessagingMetricsFeature captures messaging metrics
+    /// </summary>
+    class MessagingMetricsFeature : Feature
+    {
+        /// <inheritdoc />
+        protected internal override void Setup(FeatureConfigurationContext context)
+        {
+            var isSendOnly = context.Settings.GetOrDefault<bool>("Endpoint.SendOnly");
+            if (isSendOnly)
+            {
+                throw new Exception("Metrics are not supported on send only endpoints.");
```

Comment:
should this really throw an exception at startup or just disable the feature? 

We could use the `Prerequisite` setting via the feature ctor to automatically disable the feature when the endpoint is a send-only endpoint, would be less problematic when users have metrics enabled by default on an "endpoint configuration template" and optionally make endpoints send-only.

Reply:
We do tend to throw exceptions similar to this. e.g. in ReceiveComponent `PrepareConfiguration` method. The alternative is, log a WARN? not sure if that's very obvious and it'd cause confusion (people looking at their monitoring and not seeing the data).

Reply:
I'd say we should throw exceptions only if a misconfiguration would impact the processing behavior/correctness of the system and especially diagnostics should fail without taking down the endpoint (we've also added that to the OT guidelines). But to get this PR merged I'll raise a separate PR for this proposal so we can discuss this there without delaying the rest further